core/src/main/java/com/novoda/merlin/receiver/ConnectivityReceiver.java

```diff
@@ -1,65 +1,55 @@
 package com.novoda.merlin.receiver;
 
-import android.annotation.SuppressLint;
 import android.content.BroadcastReceiver;
 import android.content.Context;
 import android.content.Intent;
-import android.net.ConnectivityManager;
 import android.os.IBinder;
-import android.support.annotation.VisibleForTesting;
+import android.support.annotation.Nullable;
 
 import com.novoda.merlin.MerlinsBeard;
 import com.novoda.merlin.service.MerlinService;
 
 public class ConnectivityReceiver extends BroadcastReceiver {
 
-    // Lint thinks we're not calling getAction, but we actually do
-    @SuppressLint("UnsafeProtectedBroadcastReceiver")
-    @Override
-    public void onReceive(Context context, Intent intent) {
-        if (intent != null && connectivityAction(intent)) {
-            boolean isConnected = getIsConnected(context, intent);
-            String info = intent.getStringExtra(ConnectivityManager.EXTRA_EXTRA_INFO);
-            String reason = intent.getStringExtra(ConnectivityManager.EXTRA_REASON);
-            notifyMerlinService(context, ConnectivityChangeEvent.createWithNetworkInfoChangeEvent(isConnected, info, reason));
-        }
+    private final ConnectivityChangeNotifier connectivityChangeNotifier;
+
+    public ConnectivityReceiver() {
+        MerlinsBeardRetriever merlinsBeardRetriever = new MerlinsBeardRetriever() {
+            @Override
+            public MerlinsBeard getMerlinsBeard(Context context) {
```

Comment:
how about `createMerlinsBeard` and `..` what does `peekService` do ? :D

TO highlight it's going to make a new one each time, not something you might expect from a `getX` method